lib/fog/core/cache.rb

```diff
@@ -0,0 +1,130 @@
+module Fog
+  class Cache
+
+    class CacheNotFound < StandardError; end
+
+    require 'fileutils'
+    require 'yaml'
+    require 'tmpdir'
+
+    attr_reader :model
+
+    # Loads cache into memory
+    # currently on disk but can expand into diffferent cache backends if necessary.
+    #
+    # Example:
+    #   instances = Fog::Cache.load(Fog::Compute::AWS::Server, compute)
+    #   instances.first.id
+    #     => i-0569a70ae6f47d229
+    #
+    # if no cache is found, it will raise an error for handling:
+    #
+    #   rescue Fog::Cache::CacheNotFound
+    #     set_initial_cache
+    #
+    def self.load(model_klass, service)
+      cache_files = Dir.glob("#{sandbox(model_klass, service)}/*")
+
+      if cache_files.empty?
+        raise CacheNotFound
+      end
+
+      new_instances = cache_files.map { |path|
```

Comment:
If you run `rubocop -a lib/fog/core/cache.rb` it'll fix this for you, but this would read better as:

```
new_instances = cache_files.map do |path|
  model_klass = cache_data(path)[:model_klass]
  model_klass.new(cache_data(path)[:attrs])
end
```